cmd/registry/plugins/registry-lint-openapi-sample/main.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	lint "github.com/apigee/registry/cmd/registry/plugins/linter"
+	"github.com/apigee/registry/rpc"
+	"gopkg.in/yaml.v3"
+)
+
+// sampleOpenApiLintCommandExecuter is an interface through which the Sample OpenAPI linter executes.
+type sampleOpenApiLintCommandExecuter interface {
+	// Runs the sample OpenAPI linter with a provided spec and configuration path
+	Execute(specPath string, ruleIds []string) ([]*rpc.LintProblem, error)
```

Comment:
This could just be a function type - the struct seems to only exist to define this function - when needed, we could pass a function pointer instead of a struct instance.